libnmstate/ifaces/ifaces.py

```diff
@@ -71,6 +71,15 @@ def __init__(self, des_iface_infos, cur_iface_infos, save_to_disk=True):
                     raise NmstateValueError(
                         f"Duplicate interfaces names detected: {iface.name}"
                     )
+                if (
+                    iface_info.get(Interface.TYPE) == InterfaceType.UNKNOWN
+                    and iface.is_up
+                ):
+                    raise NmstateValueError(
+                        f"Interface {iface.name} is type {iface.iface_type} "
+                        "and explicitly set as up in the desired state which "
+                        "is invalid"
+                    )
```

Comment:
On second thought, I think we should not raise this exception as we support user to apply back what they get from `nmstate.show()`.

As long as we showing `unknown` interface in the `nmstate.show()`, we should allow it in desire state.

Reply:
I am fine with just ignoring that interface. Changing it :-)